app/assets/javascripts/angular/resources/shared_graph_behavior.js

```diff
@@ -60,6 +60,19 @@ angular.module("Prometheus.services").factory("SharedGraphBehavior", ["$http", "
       }
     }
 
+    function addVarsFromServerInterpolations(servers) {
+      var re = /{{\s?(\w+)\s?}}/g;
+      for (var i = 0; i < servers.length; i++) {
+        var m = re.exec(servers[i].url);
```

Comment:
I see you removed the `while`. The `re.exec` is a weird API, returning next match until it returns `null` when it can't find any more.
Without the `while`, only the first match will be found and only one interpolation will be found.